File: Medium/Java/419. Battleships in a Board.java
class Solution {
    public int countBattleships(char[][] board) {
        int count = 0;
        int r = board.length;
        int c = board[0].length;

        for(int i = 0; i < r; i++){
            for(int j = 0; j < c; j++){
                if(board[i][j] == 'X'){
                    if((i == r - 1 || board[i + 1][j] == '.') &&
                       (j == c - 1 || board[i][j + 1] == '.'))
                        count++;
                }
            }
        }

        return count;
    }
}

File: Medium/Python/419. Battleships in a Board.py
class Solution(object):
    def countBattleships(self, board):
        count = 0
        r = len(board)
        c = len(board[0])

        for i in range(r):
            for j in range(c):
                if board[i][j] == 'X':
                    if (i == r - 1 or board[i + 1][j] == '.') and (j == c - 1 or board[i][j + 1] == '.'):
                        count += 1

        return count
